### PR TITLE
ANL-1468 GraphAPIError enhancements

### PR DESCRIPTION
**Pull request by paul-sapunaru-hs**
@tudor-mihordea-hs @diana-tatu-hs please review

### CODE REVIEW

File: facebook.py [hunk 1]
```diff
@@ -305,9 +306,10 @@ def request(self, path, args=None, post_args=None):
                 args["appsecret_proof"] = self.appsecret_proof
 
         post_data = None if post_args is None else urllib.urlencode(post_args)
+        request_url = None
         try:
-            file = urllib2.urlopen(("https://graph.facebook.com/v%s/" % self.version) + path + "?" +
-                    urllib.urlencode(args), post_data, timeout=self.timeout)
+            request_url = ("https://graph.facebook.com/v%s/" % self.version) + path + "?" + urllib.urlencode(args)
+            file = urllib2.urlopen(request_url, post_data, timeout=self.timeout)
```
Comment (diana-tatu-hs) [inline facebook.py:312]: You need to set `request_url` to `''` or smth in case of a `TypeError` too

Comment (diana-tatu-hs) [inline facebook.py:312]: Otherwise, line 338 will raise `NameError: name 'request_url' is not defined`

**Comment by diana-tatu-hs** [comment]
lgtm otherwise

File: facebook.py (hunk 1 above)
Comment (paul-sapunaru-hs) [inline facebook.py:312]: It's set to `None` at line 309, but I will set it to `''` to be consistent with the default value and also add it when raising `GraphAPIError` at line 315. Thanks!

**Comment by tudor-mihordea-hs** [comment]
also update version='0.4.0.4',

**Comment by paul-sapunaru-hs** [comment]
@diana-tatu-hs @tudor-mihordea-hs please take another look before merging

File: facebook.py [hunk 2]
```diff
@@ -305,9 +306,10 @@ def request(self, path, args=None, post_args=None):
...
         except urllib2.HTTPError, e:
             response = _parse_json(e.read())
             raise GraphAPIError(response)
```
Comment (tudor-mihordea-hs) [inline facebook.py]: send the request_url here. This is our main use-case.

File: facebook.py [hunk 3]
```diff
@@ -333,8 +335,7 @@ def request(self, path, args=None, post_args=None):
         finally:
             file.close()
         if response and isinstance(response, dict) and response.get("error"):
-            raise GraphAPIError(response["error"]["type"],
-                                response["error"]["message"])
+            raise GraphAPIError(response, request_url)
```
Comment (tudor-mihordea-hs) [inline facebook.py:338]: +1 but we never reach this code now. Sentry stands as proof

**Comment by tudor-mihordea-hs** [comment]
lgtm

**Comment by diana-tatu-hs** [comment]
lgtm

**Pull request merged**